gauntlet/packages/gauntlet-solana/src/lib/contracts.ts

```diff
@@ -0,0 +1,73 @@
+import { readFileSync } from 'fs'
+import { join } from 'path'
+import { io } from '@chainlink/gauntlet-core/dist/utils'
+import { Keypair, PublicKey } from '@solana/web3.js'
+
+export type Contract = {
+  id: string
+  version: string
+  idl: any
+  programId: PublicKey
+}
+
+export type DeploymentContract = {
+  id: string
+  bytecode: Buffer
+  programKeypair: Keypair
+}
+
+export const registerContracts = <List extends Record<string, string>>(
```

Comment:
This function has still some issues, essentially: 
- Depends on an static path to read artifacts. (What if those artifacts are not in the FS?)
- Depends on the environment to load the program Ids. (What if some `gauntlet-solana` consumer chooses another way to store their ProgramIds? We would be forcing them to include it into the env)

As this is a bigger issue that require deeper changes, let's open an issue and reference this comment there.